### PR TITLE
FEATURE: Run garbage collection of configured caches

### LINKED ISSUE #863: Provide command for cache garbage collection
Provide a command controller to clean garbage (entries past lifetime) from one or all caches.

### PR DESCRIPTION
**Pull request by sorenmalling**
In order to run garbage collection the following command
has been introduced

`./flow cache:collectgarbage`

which will iterator over all configured caches and
run the corresponding `collectGarbage` method.

You can also run garbage collection on a single cache
by definined the cache identifier

`./flow cache:collectgarbage --cache-identifier Flow_Monitor`

Resolves #863

### CODE REVIEW

File: Neos.Flow/Classes/Command/CacheCommandController.php [hunk 1]
```diff
@@ -418,6 +418,39 @@ public function setupAllCommand(bool $quiet = false)
...
+            $this->outputLine('Garbage Collection for cache "%s" completed', [$cacheIdentifier]);
+        } else {
+            $cacheConfigurations = $this->cacheManager->getCacheConfigurations();
+            unset($cacheConfigurations['Default']);
+            ksort($cacheConfigurations);
+
+            foreach ($cacheConfigurations as $identifier => $configuration) {
+                $cache = $this->cacheManager->getCache($identifier);
+                $cache->collectGarbage();
+                $this->outputLine('Garbage Collection for cache "%s" completed', [$identifier]);
```
Comment (daniellienert) [inline Neos.Flow/Classes/Command/CacheCommandController.php]:
```suggestion
                $this->output('Garbage Collection for cache %s ... ', [$identifier]);
                $this->cacheManager->getCache($identifier)->collectGarbage();
                $this->outputLine('<success>completed</success>');
```

Comment (daniellienert) [reply Neos.Flow/Classes/Command/CacheCommandController.php]: Display what cache is worked on atm

File: Neos.Flow/Classes/Command/CacheCommandController.php [hunk 2]
```diff
@@ -418,6 +418,39 @@ public function setupAllCommand(bool $quiet = false)
...
+     *
+     * Though the method is defined in the BackendInterface, the implementation
+     * can differ and might not remove any data, depending on possibilities of
+     * the backend.
+     *
+     * @param string $cacheIdentifier If set, this command only applies to the given cache
+     * @return void
+     * @throws NoSuchCacheException
+     */
+    public function garbageCollectionCommand(string $cacheIdentifier = null) {
```
Comment (daniellienert) [inline Neos.Flow/Classes/Command/CacheCommandController.php]: Adjust to PSR-2 to mage Style checker happy. And please add `:void` as return type.

Comment (daniellienert) [inline Neos.Flow/Classes/Command/CacheCommandController.php]: I would recommend to name the command imperative, like `collectGarbage`

**Comment by kdambekalns** [comment]
Assigning tags I reailized this is Flow… but it's cache-related. Strange, in a way, to have that not in Flow.Cache, but off the top of my head I don't know a solution to that…

File: Neos.Flow/Classes/Command/CacheCommandController.php [hunk 3]
```diff
@@ -418,6 +418,40 @@ public function setupAllCommand(bool $quiet = false)
...
+            $cache->collectGarbage();
+
+            $this->outputLine('<success>Garbage Collection for cache "%s" completed</success>', [$cacheIdentifier]);
+        } else {
+            $cacheConfigurations = $this->cacheManager->getCacheConfigurations();
+            unset($cacheConfigurations['Default']);
+            ksort($cacheConfigurations);
+
+            foreach ($cacheConfigurations as $identifier => $configuration) {
+                $this->outputLine('Garbage Collection for cache "%s"', [$identifier]);
```
Comment (daniellienert) [inline Neos.Flow/Classes/Command/CacheCommandController.php]: A blank is missing after the cache type

**Comment by albe** [comment]
Yeah, strange, but let's get this in first and then track an issue for moving the respective CommandControllers to their respective packages in some way.

File: Neos.Flow/Classes/Command/CacheCommandController.php [hunk 4]
```diff
@@ -418,6 +418,40 @@ public function setupAllCommand(bool $quiet = false)
...
+                $cache = $this->cacheManager->getCache($identifier);
+                $cache->collectGarbage();
+                $this->outputLine('<success>Completed</success>', [$identifier]);
```
Comment (albe) [inline Neos.Flow/Classes/Command/CacheCommandController.php]:
```suggestion
                $this->outputLine('<success>Completed</success>');
```

**Comment by albe** [comment]
Gonna fix psalm errors later directly on 6.0+ branches... things have gone a bit off there.

**Pull request merged**